esphome/components/network/__init__.py

```diff
@@ -15,9 +15,7 @@
 
 CONFIG_SCHEMA = cv.Schema(
     {
-        cv.SplitDefault(CONF_ENABLE_IPV6, esp32=False): cv.All(
-            cv.only_on_esp32, cv.boolean
-        ),
+        cv.Optional(CONF_ENABLE_IPV6): cv.boolean,
```

Comment:
This should default to false for all platforms and then be utilised in `to_code` properly
```suggestion
        cv.Optional(CONF_ENABLE_IPV6, default=False): cv.boolean,
```

Reply:
Solved